### PR TITLE
fix(button): navigate menu via keyboard without moving focus

### PR DESCRIPTION
**Pull request by dtsanevmw**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- Changed the behavior of the keyboard navigation in menu to support all a11y requirements. 
- Keyboard navigation is now handled with having a current item in the menu while the focus is not changed and it's staying on the toggle button as required. This help us resolve all a11y issues which were not resolved with the previous functionality. 

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behaviour ?

- Currently the service we created was working with another "FocusService" and the navigation was happening trough focusing different items. 

Issue Number: CDE-68

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/angular/src/button/button-group/button-group.html [hunk 1]
```diff
@@ -10,21 +10,21 @@
...
       clrPopoverAnchor
       clrPopoverOpenCloseButton
-      (keydown.arrowup)="openButtonGroupMenu($event, InitialFocus.LAST_ITEM)"
-      (keydown.arrowdown)="openButtonGroupMenu($event, InitialFocus.FIRST_ITEM)"
+      (keydown.arrowup)="openMenu($event, InitialItem.LAST)"
+      (keydown.arrowdown)="openMenu($event, InitialItem.FIRST)"
       [attr.aria-controls]="popoverId"
       [attr.aria-expanded]="open"
-      [attr.aria-label]="commonStrings.keys.rowActions"
+      [attr.aria-label]="clrToggleButtonAriaLabel"
```
Comment (kevinbuhmann) [inline projects/angular/src/button/button-group/button-group.html:21]: This change seems unrelated to keyboard navigation. I would remove it from here and open a separate PR so that this change is called out as a separate feature in the change log. Otherwise, users won't know we added it unless the dig into the code.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]
See comments.

After moving the custom `aria-label` feature to a separate PR, this change could be described in the the commit message as `fix(button): navigate menu via keyboard without moving focus`. That way, it's more clear in the changelog what this does.

File: projects/angular/src/button/providers/button-group-navigation-handler.spec.ts [hunk 2]
```diff
@@ -92,32 +88,28 @@ export default function (): void {
       });
 
       it('first child is focused after open', function (this: TestContext) {
-        this.focusHandler.buttonsContainer = this.buttonsContainer;
-        this.focusHandler.menuOpened = true;
-        const firstChild = this.buttonsContainer.children[0];
-        expect(document.activeElement).toBe(firstChild);
+        this.navigationHandler.menuOpened = true;
+        const firstChild = this.menu.children[0];
+        expect(this.navigationHandler.isCurrent(firstChild.id)).toBe(true);
```
Comment (kevinbuhmann) [inline projects/angular/src/button/providers/button-group-navigation-handler.spec.ts:93]: I think this should check for the `clr-focus` class on the actual button element. That way, it's testing real behavior, not just the service implementation details.

File: projects/angular/src/button/button-group/button.ts [hunk 3]
```diff
@@ -148,6 +167,12 @@ export class ClrButton implements LoadingListener {
     this._click.emit(true);
   }
 
+  ngOnInit() {
+    if (this.inMenu && !this.id) {
+      this.id = uniqueIdFactory();
```
Comment (kevinbuhmann) [inline projects/angular/src/button/button-group/button.ts:172]: I think this can just be initialized at the `_id` property declaration. That simplifies the code and could prevent an extra DOM update.

Comment (kevinbuhmann) [reply projects/angular/src/button/button-group/button.ts:172]: It should be okay to just initialize the id of all buttons, not just the ones in the in the menu.

Comment (dtsanevmw) [reply projects/angular/src/button/button-group/button.ts:172]: I'll have to change some of the tests for button because they are validating if id is '' if it's not set.

Comment (kevinbuhmann) [reply projects/angular/src/button/button-group/button.ts:172]: Oh. An empty `id` is invalid anyway...
